### PR TITLE
[CP] [skwasm] Fix platform view occlusion logic.

### PR DESCRIPTION
**Pull request by Rexios80**
Cherry-pick of https://github.com/flutter/engine/pull/54061

### CODE REVIEW

**Review by eyebrowsoffire — APPROVED** [review]
Cherry pick looks good to me! Thanks!

**Review by eyebrowsoffire — CHANGES_REQUESTED** [review]
Looks like CI is failing, I'm guessing some there are some test changes that didn't make it over in the CP for some reason?

File: lib/web_ui/test/engine/scene_builder_test.dart [hunk 1]
```diff
@@ -217,12 +244,15 @@ class PlatformViewSliceMatcher extends Matcher {
       final PlatformView expectedView = expectedPlatformViews[i];
       final PlatformView actualView = item.views[i];
       if (expectedView.viewId != actualView.viewId) {
+        print('viewID mismatch');
         return false;
       }
-      if (expectedView.size != actualView.size) {
+      if (expectedView.bounds != actualView.bounds) {
```
Comment (Rexios80) [inline lib/web_ui/test/engine/scene_builder_test.dart:250]: @eyebrowsoffire There was a merge conflict here. Did I resolve it incorrectly?

Comment (Rexios80) [reply lib/web_ui/test/engine/scene_builder_test.dart:250]: Indeed I did

**Comment by Rexios80** [comment]
Also I did notice that there is a `flutter-3.24-candidate.1` branch. Is `flutter-3.24-candidate.0` the correct one? The https://github.com/flutter/flutter/blob/stable/bin/internal/release-candidate-branch.version file points to `flutter-3.24-candidate.0`.

**Review by eyebrowsoffire — APPROVED** [review]
Looks good! I'm not 100% sure about which branch should be targeted, I'll let the release team handle that particular question.

**Pull request merged**